Feature/ie10 fixes

apfelfabrik opened the pull request:
Implements [`#2371`](https://www.openproject.org/issues/2371)

apfelfabrik committed:
Fixes security issues around :visited in IE10.

apfelfabrik committed:
Updates deprecated method call.

apfelfabrik committed:
Moves noscript into body.

apfelfabrik committed:
Introduces HTML5 doctype.

apfelfabrik committed:
Restores now-line in timelines.

Date.DAY somehow doesn't exist anymore.

apfelfabrik committed:
Changelog.

mfrister commented:
I found an issue with IE10. When closing a modal on a multi-page timeline, manually expanded items collapse, but the browser doesn't scroll up:
Reproduce:
- Open timeline
- Expand an element resulting in the timeline growing to multiple pages
- Scroll down a few pages
- Click on a work package to open a modal
- Close the modal

Expected (assuming the collapsing was correct behavior): The timeline collapses, the browser scrolls up and the scrollbar indicates browser page as long as before expanding the elements.

Actual result: Elements collapse, the browser stays at the same position, so the shown part of the page is empty.

I'd like this to be fixed before merging this.

Also, which browsers should this change support?

apfelfabrik committed:
Fixes a scrolling problem w/ IE10.

When a timeline collapses after a modal is closed, IE10 would sometimes
still display a part of the page where there is no content. This happens
when work packages are moved around and nested so that the resulting
timeline is shorter than the previous scrollTop, or when the timeline
falls back to the default timeline expansion (a bug currently not
fixed).

apfelfabrik commented:
The bug you mentioned is fixed, thanks! The other thing, browser support, is kind of fuzzy. I believe @tessi is working on a list of software requirements.

apfelfabrik committed:
Merge branch 'feature/rails3' into feature/ie10-fixes

Conflicts:
	doc/CHANGELOG.md

Commit:
Merge remote-tracking branch 'origin/feature/rails3' into feature/ie10-fixes

Conflicts:
	doc/CHANGELOG.md

[skip ci]
